boards/arm/efr32bg_sltb010a/board.c

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2021 Sateesh Kotapati
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/init.h>
+#include <zephyr/logging/log.h>
+
+#ifdef CONFIG_SOC_GECKO_DEV_INIT
+#include <sl_device_init_clocks.h>
+#include <sl_board_init.h>
+#endif
+
+LOG_MODULE_REGISTER(efr32bg_sltb010a, CONFIG_BOARD_EFR32BG22_LOG_LEVEL);
+
+static int efr32bg_sltb010a_init(const struct device *dev)
+{
+	int ret;
+
+#ifdef CONFIG_SOC_GECKO_DEV_INIT
+	sl_device_init_clocks();
+	sl_board_init();
```

Comment:
definitely a -1 to run code like this https://github.com/antmicro/hal_silabs/blob/bg22-minimal/gecko/board/src/sl_board_init.c. This is bypassing Zephyr mechanisms, including Devicetree as a source of hardware description.

Reply:
We can actually remove this function call entirely. GPIO clock is enabled by GPIO driver and the rest of the function (peripheral init) is not needed.